tests/test_scenario/test_contacts.py

```diff
@@ -0,0 +1,226 @@
+                         ids=utils.id_gazebo_fn)
+def test_cube_multiple_contacts(gazebo: bindings.GazeboSimulator,
+                                get_model_str: Callable):
+
+    assert gazebo.initialize()
+    world = gazebo.getWorld()
+
+    # Insert the Physics system
+    assert world.insertWorldPlugin("libPhysicsSystem.so",
+                                   "scenario::plugins::gazebo::Physics")
+
+    # Insert the ground plane
+    assert world.insertModel(gym_ignition_models.get_model_file("ground_plane"))
+    assert len(world.modelNames()) == 1
+
+    # Insert two cubes side to side with a 10cm gap
+    cube_urdf = misc.string_to_file(get_model_str())
+    assert world.insertModel(cube_urdf,
+                             bindings.Pose([0, -0.15, 0.101], [1., 0, 0, 0]),
+                             "cube1")
+    assert world.insertModel(cube_urdf,
+                             bindings.Pose([0, 0.15, 0.101], [1., 0, 0, 0]),
+                             "cube2")
+    assert len(world.modelNames()) == 3
+
+    # Get the cubes
+    cube1 = world.getModel("cube1")
+    cube2 = world.getModel("cube2")
+
+    # Enable contact detection
+    assert cube1.enableContacts()
+    assert cube2.enableContacts()
+
+    # The cubes were inserted floating in the air with a 1mm gap wrt to ground.
+    # There should be no contacts.
+    gazebo.run(paused=True)
+    assert not cube1.getLink("cube").inContact()
+    assert not cube2.getLink("cube").inContact()
+    assert len(cube1.contacts()) == 0
+    assert len(cube2.contacts()) == 0
+
+    # Make the cubes fall for 50ms
+    for _ in range(50):
+        gazebo.run()
+
+    assert cube1.getLink("cube").inContact()
+    assert cube2.getLink("cube").inContact()
+    assert len(cube1.contacts()) == 1
+    assert len(cube2.contacts()) == 1
+
+    # Now we make another cube fall above the gap. It will touch both cubes.
+    assert world.insertModel(cube_urdf,
+                             bindings.Pose([0, 0, 0.301], [1., 0, 0, 0]),
+                             "cube3")
+    assert len(world.modelNames()) == 4
+
+    cube3 = world.getModel("cube3")
+    assert cube3.enableContacts()
+
+    gazebo.run(paused=True)
+    assert not cube3.getLink("cube").inContact()
+    assert len(cube3.contacts()) == 0
+
+    # Make the cube fall for 50ms
+    for _ in range(50):
+        gazebo.run()
+
+    # There will be two contacts, respectively with cube1 and cube2.
+    # Contacts are related to the link, not the collision elements. In the case of two
+    # collisions elements associated to the same link, contacts are merged together.
+    assert cube3.getLink("cube").inContact()
+    assert len(cube3.contacts()) == 2
+
+    contact1 = cube3.contacts()[0]
+    contact2 = cube3.contacts()[1]
+
+    assert contact1.bodyA == "cube3::cube"
+    assert contact2.bodyA == "cube3::cube"
+
+    assert contact1.bodyB == "cube1::cube"
+    assert contact2.bodyB == "cube2::cube"
```

Comment:
Is cube1::cube always associated with contact1 (and cube2 with contact2)?

Reply:
Since the simulation is reproducible this order is always respected. Though, I had to check beforehand because in cases like this in which contacts are created in the same moment you cannot know a priory how contacts get detected.